yarn-project/foundation/src/fields/point.ts

```diff
@@ -137,8 +137,16 @@ export class Point {
     return poseidon2Hash(this.toFields());
   }
 
+  /**
+   * Check if this is point at infinity.
+   */
+  isInfPoint() {
+    // Check this
+    return this.x.isZero();
+  }
+
   isOnGrumpkin() {
-    if (this.isZero()) {
+    if (this.isInfPoint()) {
```

Comment:
I'll trust you on this one.

Reply:
I'll pop a TODO here because I need to come back and check on this